src/apps/loyalty/server/index.tsx

```diff
@@ -72,53 +72,65 @@ app.get("/inquiries", (req, res) => {
     return res.redirect(req.baseUrl + "/login")
   }
 
-  let promise = IsomorphicRelay.prepareData({
-    Container: Inquiries,
-    queryConfig: new CurrentUserRoute(),
-  }, res.locals.networkLayer)
+  fetchCollectorProfile(req.user.get("accessToken"))
+    .then(body => body.json())
+    .then(info => {
+      if (info.loyalty_applicant_at) {
+        return res.redirect(req.baseUrl + "/thank-you")
+      }
 
-  promise
-    .then(({data, props}) => {
-      const html = renderToString(<IsomorphicRelay.Renderer {...props} />)
-      const styles = styleSheet.rules().map(rule => rule.cssText).join("\n")
-      res.send(renderPage({
-        styles,
-        html,
-        entrypoint: "/bundles/inquiries.js",
-        bootstrapData: `var DATA = ${JSON.stringify(data)}; var USER_DATA = ${JSON.stringify(req.user.toJSON())};`,
-      }))
+      if (info.confirmed_buyer_at) {
+        markCollectorAsLoyaltyApplicant(req.user.get("accessToken"))
+          .then(body => body.json())
+          .then(profile => {
+            return res.redirect(req.baseUrl + "/thank-you")
+          })
+          .catch(err => console.error(err))
+      }
+
+      let promise = IsomorphicRelay.prepareData({
+        Container: Inquiries,
+        queryConfig: new CurrentUserRoute(),
+      }, res.locals.networkLayer)
+
+      promise
+        .then(({data, props}) => {
+          const html = renderToString(<IsomorphicRelay.Renderer {...props} />)
+          const styles = styleSheet.rules().map(rule => rule.cssText).join("\n")
+          res.send(renderPage({
+            styles,
+            html,
+            entrypoint: "/bundles/inquiries.js",
+            bootstrapData: `var DATA = ${JSON.stringify(data)}; var USER_DATA = ${JSON.stringify(req.user.toJSON())};`,
+          }))
+        })
```

Comment:
Aye, it's just a style thing - not pressure to to this, would make a nice learning tool as a separate PR 👍